agent/logging_test.go

```diff
@@ -264,13 +265,70 @@ func TestLogkeeperMetadataPopulated(t *testing.T) {
 		},
 		taskModel: task,
 	}
-	assert.NoError(agt.resetLogging(ctx, tc))
+	assert.NoError(agt.startLogging(ctx, tc))
 	assert.Equal("logkeeper/build/build1/test/test1", tc.logs.AgentLogURLs[0].URL)
 	assert.Equal("logkeeper/build/build1/test/test2", tc.logs.SystemLogURLs[0].URL)
 	assert.Equal("logkeeper/build/build1/test/test3", tc.logs.TaskLogURLs[0].URL)
 	assert.Equal("", tc.logs.TaskLogURLs[0].Command)
 }
 
+func TestDefaultSender(t *testing.T) {
+	assert := assert.New(t)
+
+	agt := &Agent{
+		opts: Options{
+			HostID:       "host",
+			HostSecret:   "secret",
+			StatusPort:   2286,
+			LogPrefix:    evergreen.LocalLoggingOverride,
+			LogkeeperURL: "logkeeper",
+		},
+		comm: client.NewMock("mock"),
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	taskID := "logging"
+	taskSecret := "mock_task_secret"
+	task := &task.Task{
+		DisplayName: "task1",
+	}
+	tc := &taskContext{
+		task: client.TaskData{
+			ID:     taskID,
+			Secret: taskSecret,
+		},
+		project: &model.Project{},
+		taskConfig: &model.TaskConfig{
+			Task:         task,
+			BuildVariant: &model.BuildVariant{Name: "bv"},
+			Timeout:      &model.Timeout{IdleTimeoutSecs: 15, ExecTimeoutSecs: 15},
+		},
+		taskModel: task,
+	}
+
+	t.Run("Valid", func(t *testing.T) {
+		tc.project.Loggers = &model.LoggerConfig{}
+		tc.taskConfig.ProjectRef = &model.ProjectRef{DefaultLogger: model.BuildloggerLogSender}
+
+		assert.NoError(agt.startLogging(ctx, tc))
```

Comment:
these error's get logged as being in the other one because of the assert.New call.